app/redux/cart.js

```diff
@@ -0,0 +1,138 @@
+import axios from 'axios';
+import thunk from 'redux-thunk';
+
+// constants to be moved to a constants.js file
+const SET_ORDERS = 'SET_ORDERS';
+const ADD_TO_CART = 'ADD_TO_CART';
+const REMOVE_FROM_CART = 'REMOVE_FROM_CART';
+const EDIT_QUANTITY = 'EDIT_QUANTITY';
+
+// action creators
+export const setActiveOrderProducts = orders => {
+  return {
+    type: SET_ORDERS,
+    orders
+  };
+};
+
+//NOTE: orderItem here should be an object with the product, quantity, and price
+export const addToCart = orderItem => {
+  return {
+    type: ADD_TO_CART,
+    orderItem
+  };
+};
+
+export const removeFromCart = orderItem => {
+  return {
+    type: REMOVE_FROM_CART,
+    orderItem
+  };
+};
+
+//NOTE: orderItem here should be an object with the product, quantity, and price
+export const editQuantity = orderItem => {
+  return {
+    type: EDIT_QUANTITY,
+    orderItem
+  };
+};
+
+// thunks
+
+export const fetchOrder = activeOrder => {
```

Comment:
reconcilliation of merging guest cart and user cart is bugged

Reply:
suggestion: active users will always post to DB and NOT local storage
on sign in, if there are items in local storage:
1) if no active cart, POST active cart and POST associated cartItems from local storage
2) if active cart, POST local storage cartItems as associated to activeCart 
3) guests always use local storage (and redux store) till checkout; AT checkout, we POST a "user" for them with name, email, and address info, POST an associated order, and POST orderItems associated to that order